src/rdf-repository/grafter/rdf/repository.clj

```diff
@@ -135,13 +137,15 @@
     (.initialize)))
 
 (defn sparql-repo
-  "Given a query-url and an optional update-url String return a Sesame
-  SPARQLRepository for communicating with remote repositories."
+  "Given a query-url (String or IURI) and an optional update-url String
+  or IURI, return a Sesame SPARQLRepository for communicating with
+  remote repositories."
   ([query-url]
-     (doto (SPARQLRepository. query-url)
+     (doto (SPARQLRepository. (str query-url))
        (.initialize)))
   ([query-url update-url]
-     (doto (SPARQLRepository. query-url update-url)
+     (doto (SPARQLRepository. (str query-url)
+                              (str update-url))
```

Comment:
This may be questionable, so I'll discuss my thought process: The use of `str` makes the function more generic than just accepting URI or String, but `SPARQLRepository` itself is not very discerning.

I did have an implementation which literally checked the types like this:

	(defn coerce-url [url]
	  (cond
	    (instance? String url) url
	    (satisfies? IURL url)  (str url)
	    :else (throw (IllegalArgumentException. "URL: Will only accept a String or IURI"))))

However, you can see that we ultimate call `str` anyway and everything ends up as a string which isn't checked for correctness as a URI, in addition to there be slight performance penalties, so I canned it and went for the simple `str` call.